libr/io/p/io_fd.c

```diff
@@ -0,0 +1,99 @@
+/* radare - LGPL - Copyright 2020 - pancake */
+
+#include "r_io.h"
+#include "r_lib.h"
+#include <stdio.h>
+#include <stdlib.h>
+#include <r_cons.h>
+#include <sys/types.h>
+
+typedef struct {
+	int fd;
+	ut32 size;
+} RIOMalloc;
+
+static int __write(RIO *io, RIODesc *desc, const ut8 *buf, int count) {
+	RIOMalloc *mal = (RIOMalloc*)desc->data;
+	if (mal) {
+		return write (mal->fd, buf, count);
+	}
+	return -1;
+}
+
+static bool __resize(RIO *io, RIODesc *desc, ut64 count) {
+	RIOMalloc *mal = (RIOMalloc*)desc->data;
+	if (mal) {
+		return ftruncate (mal->fd, count) == 0;
+	}
+	return false;
+}
+
+static int __read(RIO *io, RIODesc *desc, ut8 *buf, int count) {
+	RIOMalloc *mal = (RIOMalloc*)desc->data;
+	if (mal) {
+		r_cons_break_push (NULL, NULL);
+		int res = read (mal->fd, buf, count);
+		r_cons_break_pop ();
+		return res;
+	}
+	return -1;
+}
+
+static int __close(RIODesc *desc) {
+	// dont close, could be problematic in self://
+	return 0;
+}
+
+static ut64 __lseek(RIO* io, RIODesc *desc, ut64 offset, int whence) {
+	RIOMalloc *mal = (RIOMalloc*)desc->data;
+	if (mal) {
+		return lseek (mal->fd, offset, whence);
+	}
+	return 0;
+}
+
+static bool __check(RIO *io, const char *pathname, bool many) {
+	return !strncmp (pathname, "fd://", 4);
+}
+
+static RIODesc *__open(RIO *io, const char *pathname, int rw, int mode) {
+	if (__check (io, pathname, 0)) {
+		if (r_sandbox_enable (false)) {
+			eprintf ("Do not permit fd:// in sandbox mode.\n");
+			return NULL;
+		}
+		RIOMalloc *mal = R_NEW0 (RIOMalloc);
+		if (mal) {
+			mal->fd = r_num_math (NULL, pathname + 4);
```

Comment:
this is wrong as well, it is +5. BTW, could you add a test for this plugin? As it is right now I don't think it can work, as `r_num_math` would always return 0.

Reply:
use `pathname + strlen(FD_PREFIX)`

Reply:
done